pkg/codegen/python/gen_program_test.go

```diff
@@ -22,3 +26,38 @@ func TestGenerateProgram(t *testing.T) {
 			TestCases:  test.PulumiPulumiProgramTests,
 		})
 }
+
+func TestGenerateProgramVersionSelection(t *testing.T) {
```

Comment:
Should we consider making this a part of `program_driver.go`? I don't think we want the `expectedVersion` information in a separate file for each language.

Reply:
@iwahbe I attempted but in `test.go`, this function calls`GenerateRequirements()` which imports language-specific packages that also import the `test` package, causing a circular dependency. We also need separate logic for `GenerateRequirements()` anyways similar to `GenerateProgram()`. We could centralize the `expectedVersion` information though?

Reply:
Perhaps `TestGenerateProgramVersionSelection` could call in the to the version checking framework located in the `test` package. It would pass `CheckVersion` and `GenerateRequirements` to the framework. This is the same dependency structure as `test.TestProgramCodegen`.

I want the data centralized so that when we can see and modify all our tests in one place.

I want the framework centralized so that it can be consumed by all languages (including Java and YAML).

Reply:
Refactored it by adding a few additional fields on `ProgramCodegenOptions`, we should call the existing languages' `GenerateProject()` functions instead